web-ui/src/account_recovery/new_password_form/new_password_form.js

```diff
@@ -27,22 +28,51 @@ import BackLink from 'src/common/back_link/back_link';
 import './new_password_form.scss';
 
 export class NewPasswordForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      submitButtonDisabled: true,
+      password: '',
+      errorPassword: '',
+      confirmPassword: '',
+      errorConfirmPassword: ''
+    };
+  }
+
   submitHandler = (event) => {
     event.preventDefault();
     submitForm(event, '/account-recovery', {
       userCode: this.props.userCode,
       password: this.state.password,
-      confirmation: this.state.confirmation
+      confirmPassword: this.state.confirmPassword
     }).then(() => this.props.next());
   }
 
-  handlePasswordChange = (event) => {
+  handleChangePassword = (event) => {
     this.setState({ password: event.target.value });
-  }
+    this.validatePassword(event.target.value, this.state.confirmPassword);
+  };
 
-  handlePasswordConfirmationChange = (event) => {
-    this.setState({ confirmation: event.target.value });
-  }
+  handleChangeConfirmPassword = (event) => {
+    this.setState({ confirmPassword: event.target.value });
+    this.validatePassword(this.state.password, event.target.value);
+  };
+
+  validatePassword = (password, confirmPassword) => {
+    const emptyPassword = validator.isEmpty(password);
+    const validPassword = validator.isLength(password, { min: 8, max: undefined });
```

Comment:
Why not put a maximum?
tip: Keypass allow me to generate a password with 9999 maximum.